certbot/certbot/interfaces.py

```diff
@@ -40,8 +48,192 @@ def save(self, account, client):  # pragma: no cover
         raise NotImplementedError()
 
 
-class IPluginFactory(zope.interface.Interface):
-    """IPlugin factory.
+class IConfig(zope.interface.Interface):  # pylint: disable=inherit-non-class
+    """Deprecated, use certbot.interfaces.Config as ABC instead."""
+
+
+@zope.interface.implementer(IConfig)
+class Config(metaclass=ABCMeta):
```

Comment:
I looked at the discussion, and I agree that having a interface for `NamespaceConfig` is giving more constraints that is usefulness regarding API contracts. I followed your proposition about moving out `NamespaceConfig` in the public API so that it can be used by third parties.

For `config_help`, I essentially reused the same approach and introspect the docstring of `NamespaceConfig` instead of `Config`. Out of curiosity, which kind of heavy refactoring to you have in mind, another way than this kind of introspection?

Reply:
> Out of curiosity, which kind of heavy refactoring to you have in mind, another way than this kind of introspection?

I didn't have something in particular in mind. I just don't love having to define all of these properties when `__getattr__` does the same thing. These properties may be good for `mypy`'s benefit anyway though.

If we wanted to do something else, my first thought would be to store the list of parameters and their documentation somewhere and use that both for `config_help` and for providing reasonable documentation for the API. I personally think the approach you took is just fine though.

Reply:
Ok!